flink-dist/pom.xml

```diff
@@ -588,6 +588,31 @@ under the License.
 				</executions>
 			</plugin>
 
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-antrun-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>package</id>
+						<phase>package</phase>
+						<goals>
+							<goal>run</goal>
+						</goals>
+						<configuration>
+							<target>
+								<delete>
+									<fileset dir="${project.basedir}/../flink-python/pyflink" includes="**/*.pyc"/>
```

Comment:
whenever you have to move into other modules like this it is a good sign that you're doing something that you really shouldn't. We've run into _so_ many issues due to things like this that I'm inclined to reject this PR.

Reply:
I mean that this logic should continue to be encapsulated in flink-python. Flink-dist should only do 1 thing: pick assembled artifacts and copy them somewhere.

What problem are we solving here? What's so bad about having a separate pom in flink-python?

Reply:
The batch Python API for example bundles the python classes as a zip in the jar; which has the nice advantage that flink-dist doesn't even know about this.

Reply:
Ohh, My initial idea was to put it in` flink-python`, and I think there's only one motivation for this change: to make `flink-python` purer. But I agree that `flink-dist` also has a pure responsibility for only copying the files it needs. 

What do you think @dianfu

Reply:
The original motivation of this PR is that pom is a Java kind of things and we want to get rid of it in flink-python module and make flink-python module a pure Python module. But considering that this may introduce another problem just as @zentol are concerned, I'm fine to keep things there it is.